src/Plugin/views/field/CollaboraPreview.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\collabora_online\Plugin\views\field;
+
+use Drupal\collabora_online\Cool\CoolUtils;
+use Drupal\Core\Url;
+use Drupal\views\Attribute\ViewsField;
+use Drupal\views\Plugin\views\field\LinkBase;
+use Drupal\views\ResultRow;
+
+/**
+ * Field handler for link to preview a collabora file.
+ *
+ * @ingroup views_field_handlers
+ */
+#[ViewsField('media_collabora_preview')]
+class CollaboraPreview extends LinkBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getUrlInfo(ResultRow $row): Url|null {
+    /** @var \Drupal\media\MediaInterface $entity */
+    $entity = $this->getEntity($row);
+
+    if ($entity === NULL) {
+      return NULL;
+    }
+
+    return CoolUtils::getEditorUrl($entity, FALSE);
+  }
+
+}
```

Comment:
I notice we need a default value for the "Text to display".
Currently when adding the field it just says "link" as default.
I propose "View in Collabora Online", to align with the entity operations.
(same for edit)

Reply:
Good catch, thanks.

Adding little modification to test default labels on test.